platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java

```diff
@@ -54,25 +69,91 @@
 public class AttributeSharingHashSessionIdManager extends HashSessionIdManager {
   // changed this to extend Jetty's impl instead of the abstract class because of a hack in pax web
 
-  private final Map<String, Set<WeakReference<HttpSession>>> sessions = new ConcurrentHashMap<>();
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AttributeSharingHashSessionIdManager.class);
+
+  static class SharingSessionInvalidator implements HttpSessionInvalidator {
+
+    private final AttributeSharingHashSessionIdManager idManager;
+
+    SharingSessionInvalidator(AttributeSharingHashSessionIdManager idManager) {
+      this.idManager = idManager;
+    }
+
+    @Override
+    public void invalidateSession(
+        String subjectName, Function<HttpSession, String> sessionSubjectExtractor) {
+
+      final List<AbstractSession> matchingSessions =
+          getHttpSessionStream()
+              .filter(s -> subjectName.equals(sessionSubjectExtractor.apply(s)))
+              .filter(AbstractSession.class::isInstance)
+              .map(AbstractSession.class::cast)
+              .distinct()
+              .collect(Collectors.toList());
+      for (AbstractSession matchingSession : matchingSessions) {
+        idManager.removeSession(matchingSession);
+        if (matchingSession.isValid()) {
+          matchingSession.invalidate();
+        }
+      }
+    }
+
+    private Stream<HttpSession> getHttpSessionStream() {
+      return getFullSessions()
+          .stream()
+          .flatMap(Collection::stream)
+          .map(Reference::get)
+          .filter(Objects::nonNull);
+    }
+
+    private Collection<Set<WeakReference<HttpSession>>> getFullSessions() {
+      return Collections.unmodifiableCollection(idManager.sessionMap.values());
+    }
+  }
+
+  private final Map<String, Set<WeakReference<HttpSession>>> sessionMap = new ConcurrentHashMap<>();
+
+  @SuppressWarnings("squid:S1149" /* Dictionary by contract */)
+  private void registerSessionManager() {
+    Bundle bundle = FrameworkUtil.getBundle(AttributeSharingHashSessionIdManager.class);
+    if (bundle == null) {
+      LOGGER.error("Error initializing Session Manager");
+      return;
+    }
+    final BundleContext bundleContext = bundle.getBundleContext();
```

Comment:
Probably wouldn't ever happen but bundle.getBundleContext() can return null.

Reply:
I'll handle that case as well.